src/methods/extract.jl

```diff
@@ -311,50 +311,44 @@ end
 
 # _rowtype returns the complete NamedTuple type for a point row
 # This code is entirely for types stability and performance.
-_rowtype(x, g; kw...) = _rowtype(x, typeof(g); kw...)
-_rowtype(x, g::Type; geometry, index, skipmissing, names, kw...) = 
-    _rowtype(x, g, geometry, index, skipmissing, names)
-function _rowtype(x, ::Type{G}, geometry::_True, index::_True, skipmissing::_True, names::NamedTuple{Names}) where {G,Names}
-    keys = (:geometry, :index, Names...,) 
-    types = Tuple{G,Tuple{Int,Int},_nametypes(x, names, skipmissing)...}
-    NamedTuple{keys,types}
+function _rowtype(x, g; geometry, index, skipmissing, names, kw...)
+    G = if skipmissing isa _True
+        nonmissingtype(typeof(g))
+    else
+        typeof(g)
+    end
+    _rowtype(x, G; geometry, index, skipmissing, names)
 end
-function _rowtype(x, ::Type{G}, geometry::_True, index::_False, skipmissing::_True, names::NamedTuple{Names}) where {G,Names}
-    keys = (:geometry, Names...,)
-    types = Tuple{G,_nametypes(x, names, skipmissing)...}
+function _rowtype(x, g::Type; geometry, index, skipmissing, names, kw...)
+    keys = _rowkeys(geometry, index, names)
+    types = _rowtypes(x, g, geometry, index, skipmissing, names)
```

Comment:
```suggestion
function _rowtype(x, ::Type{G}; geometry, index, skipmissing, names, kw...)
    keys = _rowkeys(geometry, index, names)
    types = _rowtypes(x, G, geometry, index, skipmissing, names)
```

Don't give julia the option of not propagating G as a known type